mono/metadata/file-mmap-posix.c

```diff
@@ -525,8 +539,10 @@ mono_mmap_map (void *handle, gint64 offset, gint64 *size, int access, void **mma
 
 	mmap_offset = align_down_to_page_size (offset);
 	eff_size += (offset - mmap_offset);
+	MONO_ENTER_GC_SAFE;
 	//FIXME translate some interesting errno values
-	res.address = mono_file_map ((size_t)eff_size, acess_to_mmap_flags (access), fh->fd, mmap_offset, &res.free_handle);
```

Comment:
Does this actually do any I/O?

Reply:
It calls mmap internally

Reply:
Which doesn't do I/O usually. Reads are deferred until you touch memory. Writes are deferred until there is memory pressure. Except on unusual systems w/o paging which will do the reading up front.

Reply:
Even without I/O, it should use a bunch of locks, so again I've opted to leave the transition in just in case